client/src/components/Dashboard/index.tsx

```diff
@@ -205,6 +207,13 @@ export default class Dashboard extends React.Component<IDashboardProps, IDashboa
     this.setState({ askDownload: true, downloadFiles: downloadFiles });
   }
 
+  onDownloadDashboard() {
+    let { dashboard } = this.props;
+    dashboard.config.layout.layouts = dashboard.config.layout.layouts || {};
+    let stringDashboard = ConfigurationsActions.convertDashboardToString(dashboard);
+    downloadBlob('return ' + dashboard, 'application/json', dashboard.id.replace(' ','_') + '.private.js');
```

Comment:
its not replace(' ','') its replace(' ','_'), because if by any chance the user put space i dont want a space in the file name